Sources/TensorFlow/Core/TensorShape.swift

```diff
@@ -73,43 +73,78 @@ public struct TensorShape: ExpressibleByArrayLiteral {
     }
 }
 
-public extension TensorShape {
+extension TensorShape: Collection, MutableCollection, RandomAccessCollection, RangeReplaceableCollection {
```

Comment:
This is minor, but would you mind separating conformances to different protocols into separate extensions? Each conformance extension should define just the requirement implementations for that conformance : for example, `extension TensorShape: RandomAccessCollection { .. }` should contain just `RandomAccessCollection` requirement implementations. This "correct separation" isn't enforced by the compiler, so please take care!

Motivation: separating conformances is good for separation of concerns.

Reply:
@dan-zheng, I left Collection and MutableCollection together in single extension block, MutableCollection requires same subscript methods. RandomAccessCollection and RangeReplaceableCollection are now separate extensions.

Reply:
> @dan-zheng, I left Collection and MutableCollection together in single extension block, MutableCollection requires same subscript methods.

Ah, I see. `MutableCollection` requires `subscript` with additional `set` accessor? If so, that sounds reasonable to me! Thanks for the context.

Reply:
> Ah, I see. MutableCollection requires subscript with additional set accessor?

That's correct!